opp_path_selection/src/path_selection_artist.cpp

```diff
@@ -0,0 +1,487 @@
+    mesh_msg.triangles[i].vertex_indices[2] = pcl_mesh.polygons[i].vertices[2];
+  }
+
+  return true;
+}
+
+bool pclFromShapeMsg(const shape_msgs::Mesh& mesh_msg, pcl::PolygonMesh& pcl_mesh)
+{
+  // Make sure that there are at least three points and at least one polygon
+  if (mesh_msg.vertices.size() < 3 || mesh_msg.triangles.size() < 1)
+  {
+    return false;
+  }
+
+  // Prepare PCL structures to receive data
+  // Resizing once now saves time later
+  pcl::PointCloud<pcl::PointXYZ> vertices;
+  vertices.resize(mesh_msg.vertices.size());
+  pcl_mesh.polygons.resize(mesh_msg.triangles.size());
+
+  // Copy the vertex indices (which describe each polygon) from
+  // the old mesh to the new mesh
+  for (std::size_t i = 0; i < mesh_msg.vertices.size(); ++i)
+  {
+    vertices[i]._PointXYZ::x = static_cast<float>(mesh_msg.vertices[i].x);
+    vertices[i]._PointXYZ::y = static_cast<float>(mesh_msg.vertices[i].y);
+    vertices[i]._PointXYZ::z = static_cast<float>(mesh_msg.vertices[i].z);
+  }
+
+  // Copy the vertex indices (which describe each polygon) from
+  // the old mesh to the new mesh
+  for (std::size_t i = 0; i < mesh_msg.triangles.size(); ++i)
+  {
+    pcl_mesh.polygons[i].vertices.resize(3);
+    pcl_mesh.polygons[i].vertices[0] = mesh_msg.triangles[i].vertex_indices[0];
+    pcl_mesh.polygons[i].vertices[1] = mesh_msg.triangles[i].vertex_indices[1];
+    pcl_mesh.polygons[i].vertices[2] = mesh_msg.triangles[i].vertex_indices[2];
+  }
+
+  // Use the filled pointcloud to populate the pcl mesh
+  pcl::toPCLPointCloud2(vertices, pcl_mesh.cloud);
+
+  return true;
+}
+
+}  // namespace
+
+namespace YAML
+{
+template <>
+struct convert<opp_path_selection::PathSelectorParameters>
+{
+  static Node encode(const opp_path_selection::PathSelectorParameters& rhs)
+  {
+    Node node;
+    node["line_threshold"] = rhs.line_threshold;
+    return node;
+  }
+
+  static bool decode(const Node& node, opp_path_selection::PathSelectorParameters& rhs)
+  {
+    if (node.size() != 1)
+    {
+      return false;
+    }
+    rhs.line_threshold = node["line_threshold"].as<decltype(rhs.line_threshold)>();
+    return true;
+  }
+};
+
+}  // namespace YAML
+
+namespace opp_path_selection
+{
+PathSelectionArtist::PathSelectionArtist(const ros::NodeHandle& nh,
+                                 const std::string& world_frame,
+                                 const std::string& sensor_frame)
+  : nh_(nh), world_frame_(world_frame), sensor_frame_(sensor_frame)
+{
+  marker_pub_ = nh_.advertise<visualization_msgs::MarkerArray>(MARKER_ARRAY_TOPIC, 1, false);
+  listener_.reset(new tf::TransformListener(nh_));
+
+  if (!listener_->waitForTransform(sensor_frame_, world_frame_, ros::Time(0), ros::Duration(TF_LOOKUP_TIMEOUT)))
+  {
+    ROS_ERROR("Transform lookup from %s to %s timed out", sensor_frame_.c_str(), world_frame_.c_str());
+    throw std::runtime_error("Transform lookup timed out");
+  }
```

Comment:
Why not just handle the failure case when you go to look up the transformation? Maybe the transformation will exist at that time. I've personally stopped doing checks like this because it's easier to handle the failure where it happens rather than trying to preempt it with a check like this